api/catalog/api/controllers/search_controller.py

```diff
@@ -506,14 +506,28 @@ def _get_result_and_page_count(
     :param results: The list of filtered result Hits.
     :return: Result and page count.
     """
-    if results is None:
-        return 0, 1
+    if not results:
+        return 0, 0
+
+    if page == 0:
```

Comment:
Honestly, this even seems redundant since we have the validators that prevent the users from ever setting the `page` to 0, and if you send the request with `page=0`, you get a 400 response: https://api.openverse.engineering/v1/images/?q=nature&page=0.

Reply:
That's true. I had mixed feelings about it. Do you think I should instead update the tests to only pass inputs we know would be valid according to the API? That doesn't seem unreasonable, really...

Reply:
If we leave it as is, then it'll be very easy to allow for `page=0` in the API request in the future if we ever decide to do that. However, it will also mean that an unnecessary clause will run for every request. Even though it's very (extremely) short. I'm also not sure, but since we're not likely to change the allowed pages,, it seems better to save a tiny amount of resources on each request :)

Reply:
I changed it to remove the impossible edge case, thanks for the feedback :pray: